arch/arc/core/prep_c.c

```diff
@@ -66,6 +67,36 @@ static void invalidate_dcache(void)
 }
 #endif
 
+#ifdef CONFIG_ISA_ARCV3
```

Comment:
ARC Cluster (CLN) may not be just ARCV3, also ARC-V. For now this guard is ok, but is it needed?
Moreover, why the dynamic discovery and checks in this scm_enable function?
Wouldn't it be more 'Zephyr' to define the hardware configuration in a DTS entry, and use that for conditional compilation?
Guess codesize is less of a concern on ARCV3 cores, but still I'd prefer DTS-driven conditional compilation instead of always calling 'enable' and just doing nothing if there is no 2nd level cache.

Reply:
> ARC Cluster (CLN) may not be just ARCV3, also ARC-V.

That's true, hovever we won't use this code base for ARC-V. End even if the same component will be reused for ARC-V it would have different interface.

IOW: it's only for ARCv3.